string node_name
string[] devices
bool[] active
